tests/Integration/PrestaShopBundle/Controller/Admin/SurvivalTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration\PrestaShopBundle\Controller\Admin;
+
+use Tests\Integration\PrestaShopBundle\Test\WebTestCase;
+
+/**
+ * @group demo
+ */
+class SurvivalTest extends WebTestCase
+{
+    /**
+     * @dataProvider getDataProvider
+     * @param $pageName
+     * @param $route
+     */
+    public function testPagesAreAvailable($pageName, $route)
+    {
+        $this->client->request(
+            'GET',
+            $this->router->generate(
+                $route
+            )
+        );
+
+        $response = $this->client->getResponse();
+
+        if ($response->isServerError()) {
+            file_put_contents($route. '.html', $response->getContent());
```

Comment:
Why writing something on disk?

Reply:
If it's in case it failed, it's useless, a test shouldn't failed :D 
Otherwise we need to do exactly the same thing for every tests.

Reply:
Maybe we can comment this block, but doing this is the first step to be able to understand why the test is failing.

Reply:
Imagine you have a problem with your installation, you ran all tests before going to bathroom (so much coffee), you came back, and hop look like you have the same size of directory as node_modules :D 
Really, it isn't a good idea to saved it in a file, every developpers have their own way to debug. Otherwise you can play with monolog verbosity and phpunit verbosity.

Reply:
It can't happen to me, I never run the tests! :trollface: 

I remove this block, I'll add a comment instead.